File: Java_Completo_Programacao_Orientada_a_Objetos_Nelio Alves/Secao_14_Heranca_e_Polimorfismo/Metodos_Abstratos/src/entities/Circle.java
package entities;

import entities.enums.Color;

public class Circle extends Shape{

	private Double radius;
	
	public Circle() {
		super();
	}
	
	public Circle(Color color, Double radius) {
		super(color);
		this.radius = radius;
	}
	
	public Double getRadius() {
		return radius;
	}

	public void setRadius(Double radius) {
		this.radius = radius;
	}

	//Herdando o metodo da classe abstrata Shape/forma,
	//E adicionando o próprio calculo da area do circulo.
	@Override
	public double area() {
		return Math.PI * radius * radius;
	}

}

File: C_Sharp_completo_programacao_orientada_a_objetos/Secao-10-Aula-145-Exercicio-resolvido/Secao-10-Aula-145-Exercicio-resolvido/Entities/Circle.cs
using System;
using Secao_10_Aula_145_Exercicio_resolvido.Entities.Enums;

namespace Secao_10_Aula_145_Exercicio_resolvido.Entities
{
    internal class Circle : Shape
    {
        public double Radius { get; set; }

        public Circle(double radius, Color color) : base(color)
        {
            Radius = radius;
        }

        public override double Area()
        {
            return Math.PI * Radius * Radius;
        }
    }
}
